ngraph/python/src/pyngraph/discrete_type_info.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pyngraph/discrete_type_info.hpp"
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+#include "ngraph/type.hpp"
+
+namespace py = pybind11;
+
+void regclass_pyngraph_DiscreteTypeInfo(py::module m) {
+    py::class_<ngraph::DiscreteTypeInfo, std::shared_ptr<ngraph::DiscreteTypeInfo>> discrete_type_info(
+        m,
+        "DiscreteTypeInfo");
+    discrete_type_info.doc() = "ngraph.impl.DiscreteTypeInfo wraps ngraph::DiscreteTypeInfo";
+    discrete_type_info.def(
+        "__lt__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version < other.version || (self.version == other.version && strcmp(self.name, other.name) < 0);
+        },
+        py::is_operator());
+    discrete_type_info.def(
+        "__le__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version < other.version ||
+                   (self.version == other.version && strcmp(self.name, other.name) <= 0);
+        },
+        py::is_operator());
+    discrete_type_info.def(
+        "__gt__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version < other.version || (self.version == other.version && strcmp(self.name, other.name) > 0);
+        },
+        py::is_operator());
+    discrete_type_info.def(
+        "__ge__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version < other.version ||
+                   (self.version == other.version && strcmp(self.name, other.name) >= 0);
+        },
+        py::is_operator());
+    discrete_type_info.def(
+        "__eq__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version == other.version && strcmp(self.name, other.name) == 0;
+        },
+        py::is_operator());
+    discrete_type_info.def(
+        "__ne__",
+        [](const ngraph::DiscreteTypeInfo& self, const ngraph::DiscreteTypeInfo& other) {
+            return self.version != other.version || strcmp(self.name, other.name) != 0;
+        },
+        py::is_operator());
+    discrete_type_info.def("__repr__", [](const ngraph::DiscreteTypeInfo& self) {
+        std::string name = std::string(self.name);
+        std::string version = std::to_string(self.version);
+        std::string parent_name = self.parent != nullptr ? self.parent->name : "None";
+        std::string parent_version = self.parent != nullptr ? std::to_string(self.parent->version) : "";
+
+        return "<'" + name + version + " Parent('" + parent_name + parent_version + "')" + "'>";
```

Comment:
I would like to see `__repr__` return something in style of: `<DiscreteTypeInfo: [OpName] v[OpVersion] Parent([ParentName] v[ParentVersion])>`, please keep spaces between version etc

Also if Parent is `nullptr` I would consider not giving the Parent's info at all - but this is not required feature.

Reply:
Updated.